### ISSUE TITLE
apollo-server-plugin-response-cache "get" call even if no @cacheControl specified

### ISSUE THREAD

**Issue by rubnogueira** [apollographql/apollo-server#6615]
`apollo-server-plugin-response-cache` makes a "get" call even if no `@cacheControl` is specified, slowing down GraphQL queries and mutations, and using resources unnecessarily.

## Explanation
`apollo-server-plugin-response-cache` uses two [Request lifecycle events](https://www.apollographql.com/docs/apollo-server/integrations/plugins-event-reference/#request-lifecycle-events). 

`willSendResponse`: for setting the cache.
`responseForOperation`: for getting the cache.

When I make a call that uses the @cacheControl directive, the `requestContext.overallCachePolicy` parameter from the `responseForOperation` doesn't contain the maxAge and scope values. Contents:

```json
{
  maxAge: undefined,
  scope: undefined,
  restrict: [Function: restrict],
  replace: [Function: replace],
  policyIfCacheable: [Function: policyIfCacheable]
}
```

According to the documentation:
```The responseForOperation event is fired immediately before GraphQL execution would take place.```
While I understand that `responseForOperation` is not supposed to contain that information, the `willSendResponse` was the only hook that returned the maxAge and scope information (I tried all the other hooks).

## Questions
- Is it possible to expose the maxAge and scope to the responseForOperation? 
- Does it make sense to use another hook for the response cache plugin to handle to avoid this call?

## Versions
apollo-server: 3.9.0
apollo-server-plugin-response-cache: 3.6.1

Thanks!

**Comment by rubnogueira** [comment]
I got your point @glasser. We don't have plans to use dynamic cache control in the resolvers, so I guess a possible solution is to implement something like this:

```js
return new ApolloServer({
    dataSources,
    schema,
    cache,
    plugins: [
      responseCachePlugin({
        shouldReadFromCache: requestContext => {
         // If found some @cachecontrol implemented in the call (types, etc.) return false 
         // Otherwise return true
         return true
        }
      })
    ]
  });
```

**Comment by harm-less** [comment]
Today I ran into this issue as well. I'd have expected a config flag like `applyGlobally` to enable the caching for the entire api or not. Or I can image a flag like `context.overallCachePolicy.requestHasCustomCacheControls` or a `context.overallCachePolicy.customCacheControls` array containing all locations of the `cacheControl` usages and it's arguments (which would be useful for debugging purposes too btw).

I came up with this solution. Considering `maxAge` is always 0, for our intends and purposes all request are properly resolved:

```ts
const plugins = [
  ApolloServerPluginCacheControl()
  responseCachePlugin({
    shouldWriteToCache: (context) => {
      const maxAge = context.overallCachePolicy.maxAge;
      return maxAge ? maxAge > 0 : false;
    }
  })
]
```
